### PR TITLE
feat: Add helper function for creating golden tests

### PR DESCRIPTION
**Pull request by st-pasha**
# Description

Added new function `testGolden()` that reduces the boilerplate of writing golden tests to a minimum.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [-] I have updated/added relevant examples in `examples`.

## Breaking Change

- [-] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: packages/flame_test/lib/src/test_golden.dart [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+/// then the test framework renders your game widget into an image, and compares
+/// that image against stored [goldenFile]. The test passes if two images are
+/// identical, or fails if the images differ even in a single pixel.
+///
+/// The term _golden file_ refers to the true rendering of a given game scene,
+/// captured at the creation of the test. In order to create a golden file, you
+/// first specify its desired name in the [goldenFile] parameter, and then run
+/// the tests using the command
+/// ```
+/// flutter test --update-goldens
```
Comment (spydon) [inline packages/flame_test/lib/src/test_golden.dart:17]: Would be nice to add this to melos too, so that you can run `melos update-goldens` from anywhere.

**Review by spydon — APPROVED** [review]
Nice!

**Review by wolfenrain — APPROVED** [review]

File: packages/flame_test/lib/src/test_golden.dart (hunk 1 above)
Comment (wolfenrain) [reply packages/flame_test/lib/src/test_golden.dart:17]: Agreed!

**Comment by spydon** [comment]
The scope that you add in the conventional commit tag doesn't have to be the package, Melos already figures that out. The scope is more to group things belonging to the same epic I think, like the text one you created.

File: packages/flame_test/lib/src/test_golden.dart (hunk 1 above)
Comment (st-pasha) [reply packages/flame_test/lib/src/test_golden.dart:17]: added

**Pull request merged**